api/src/pcapi/static/backoffice/js/addons/pc-offer-criterion-form.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Enables or disables highlight selection depending on selected categories.
+ */
+class PcOfferCriterionForm extends PcAddOn {
+  static CATEGORIES_SELECTOR = 'form[name="tags_create"] .pc-field-categories select, form[name^="tags_"][name$="_update"] .pc-field-categories select';
+  static HIGHLIGHT_SELECTOR = '.pc-field-highlight'
+
+
+  initialize = () => {
+    const $categoriesSelect = document.querySelector(PcOfferCriterionForm.CATEGORIES_SELECTOR)
+    if ($categoriesSelect) {
+      this._categoriesChange({target: $categoriesSelect})
+    }
+  }
+
+  bindEvents = () => {
+    EventHandler.on(document.body, 'change', PcOfferCriterionForm.CATEGORIES_SELECTOR, this._categoriesChange)
+
+    const callback = (mutationsList, observer) => {
+      for (const mutation of mutationsList) {
+        if (mutation.type === 'childList') {
+          mutation.addedNodes.forEach(node => {
+            if (node.nodeType !== 1) return;  // nodeType === 1 if HTML element
+            const $categoriesSelect = node.matches(PcOfferCriterionForm.CATEGORIES_SELECTOR) ? node : node.querySelector(PcOfferCriterionForm.CATEGORIES_SELECTOR);
+            if ($categoriesSelect) {
+              this._categoriesChange({target: $categoriesSelect})
+              return;
+            }
+          });
+        }
+      }
+    };
+
+  const observer = new MutationObserver(callback);
+  observer.observe(document.body, { childList: true, subtree: true });
```

Comment:
Détail d'indentation
```suggestion
    const observer = new MutationObserver(callback);
    observer.observe(document.body, { childList: true, subtree: true });
```